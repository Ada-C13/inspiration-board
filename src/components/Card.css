.card {
  background-color: #F4FF81;

  padding: 1em 0;
  margin: 0.5rem;

  width: 350px;
  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  align-self: start;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  cursor: pointer;
  margin-right: 10px;
  border: 0;
  border-radius: 3px;
  letter-spacing: 0.1em;
  background: transparent;
  border: 2px solid #000;
  z-index: 2;
}

.card__delete:hover {
  color: rgba(145, 22, 22, 0.788);
  border: 2px solid rgba(145, 22, 22, 0.788);
}
