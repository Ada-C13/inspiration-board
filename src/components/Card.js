import React, { Component } from "react";
import PropTypes from "prop-types";
import emoji from "emoji-dictionary";

import "./Card.css";

const Card = (props) => {
  return (
    <div className="card">
      Message: <br />
      {props.text} <br />
      Emoji: <br />
      {props.emoji} <br />
    </div>
  );
};

Card.propTypes = {
  message: PropTypes.string.isRequired,
  emoji: PropTypes.string,
};

export default Card;
