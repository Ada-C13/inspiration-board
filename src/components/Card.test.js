import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Card from './Card';

describe('Card', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Card
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});