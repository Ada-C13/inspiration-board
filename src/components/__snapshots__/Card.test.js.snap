// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <section
      class="card__content"
    >
      <p
        class="card__content-text"
      >
        You can do it!
      </p>
      <p
        class="card__content-emoji"
      >
        ❤️
      </p>
    </section>
    <button
      class="card__delete"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;
